hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoAdjustAllowanceTransitionLogic.java

```diff
@@ -189,37 +200,33 @@ private void adjustNftAllowances(final List<NftAllowance> nftAllowances, final A
 			final var owner = allowance.getOwner();
 
 			final var accountToAdjust = fetchOwnerAccount(owner, payerAccount, accountStore, entitiesChanged);
-			final var nftAllowancesMap = accountToAdjust.getMutableNftAllowances();
+			final var mutableApprovedForAllNftsAllowances = accountToAdjust.getMutableApprovedForAllNftsAllowances();
 
 			final var spenderAccount = allowance.getSpender();
 			final var approvedForAll = allowance.getApprovedForAll();
 			final var serialNums = allowance.getSerialNumbersList();
 			final var tokenId = allowance.getTokenId();
 			final var spender = Id.fromGrpcAccount(spenderAccount);
 			accountStore.loadAccountOrFailWith(spender, INVALID_ALLOWANCE_SPENDER_ID);
-
 			final var key = FcTokenAllowanceId.from(EntityNum.fromTokenId(tokenId),
 					spender.asEntityNum());
-			if (!nftAllowancesMap.containsKey(key)) {
-				final var value = approvedForAll.getValue() ? FcTokenAllowance.from(
-						true) : FcTokenAllowance.from(serialNums);
-				nftAllowancesMap.put(key, value);
+
+			if (approvedForAll.getValue()) {
+				mutableApprovedForAllNftsAllowances.add(key);
 			} else {
-				final var oldValue = nftAllowancesMap.get(key);
-				if (approvedForAll.getValue()) {
-					nftAllowancesMap.put(key, FcTokenAllowance.from(true));
-				} else {
-					final var newSerials = adjustSerials(oldValue.getSerialNumbers(), serialNums);
-					if (newSerials.isEmpty()) {
-						nftAllowancesMap.remove(key);
-					} else {
-						nftAllowancesMap.put(key, FcTokenAllowance.from(newSerials));
-					}
-				}
+				mutableApprovedForAllNftsAllowances.remove(key);
+			}
+
+			final var nfts = new ArrayList<UniqueToken>();
+			for (var serialNum : serialNums) {
```

Comment:
‼️ I may be overlooking something, but surely we need to validate that this NFT is owned by the granting account? 😀